operator-common/src/main/java/io/strimzi/operator/common/controller/SimplifiedReconciliation.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.controller;
+
+import io.strimzi.operator.common.Reconciliation;
+
+/**
+ * This simplified class is used initially instead of the regular Reconciliation class. It also has a custom equals
+ * implementation to detect if the same resource is already enqueued or not. It doesn't yet request the
+ * reconciliation ID. Not issuing the reconciliation ID right away makes the IDs more linear and means they are not
+ * requested unless a reconciliation really starts.
+ */
+public class SimplifiedReconciliation {
```

Comment:
I do not want to do that - I think it could be confusing given how we override the equls method for our particular use-case. TBH, ideally, I wanted to avoid the name referring to the `Reconciliation` class as that is a bit confusing on its own. But I didn't managed to come up with a better name and some fields are common with the full `Reconciliation` class.